freqtrade/exchange/exchange.py

```diff
@@ -2517,6 +2538,156 @@ async def _fetch_funding_rate_history(
+            ):
+                logger.debug(f"Refreshing TRADES data for {pair}")
+                # fetch trades since latest _trades and
+                # store together with existing trades
+                try:
+                    until = None
+                    from_id = None
+                    if is_in_cache:
+                        from_id = self._trades[(pair, timeframe, candle_type)].iloc[-1]["id"]
+                        until = dt_ts()  # now
+
+                    else:
+                        until = int(timeframe_to_prev_date(timeframe).timestamp()) * 1000
+                        all_stored_ticks_df = data_handler.trades_load(f"{pair}", self.trading_mode)
+
+                        if not all_stored_ticks_df.empty:
+                            if all_stored_ticks_df.iloc[0]["timestamp"] <= first_candle_ms:
+                                last_cached_ms = all_stored_ticks_df.iloc[-1]["timestamp"]
+                                # only use cached if it's closer than first_candle_ms
+                                since_ms = (
+                                    last_cached_ms
+                                    if last_cached_ms > first_candle_ms
+                                    else first_candle_ms
+                                )
+                            # doesn't go far enough
+                            else:
+                                all_stored_ticks_df = DataFrame(
+                                    columns=DEFAULT_TRADES_COLUMNS + ["date"]
+                                )
+
+                    # from_id overrules with exchange set to id paginate
+                    [_, new_ticks] = self.get_historic_trades(
+                        pair,
+                        since=since_ms if since_ms else first_candle_ms,
+                        until=until,
+                        from_id=from_id,
+                    )
+
+                except Exception as e:
+                    logger.error(f"Refreshing TRADES data for {pair} failed")
+                    logger.error(e)
+                    raise e
+
+                if new_ticks:
+                    all_stored_ticks_list = all_stored_ticks_df[
+                        DEFAULT_TRADES_COLUMNS
+                    ].values.tolist()
+                    all_stored_ticks_list.extend(new_ticks)
+                    trades_df = self._process_trades_df(
+                        pair,
+                        timeframe,
+                        candle_type,
+                        all_stored_ticks_list,
+                        cache,
+                        first_required_candle_date=first_candle_ms,
+                    )
+                    results_df[(pair, timeframe, candle_type)] = trades_df
+                    data_handler.trades_store(
+                        f"{pair}", trades_df[DEFAULT_TRADES_COLUMNS], self.trading_mode
+                    )
+
+                else:
+                    raise OperationalException("no new ticks")
```

Comment:
You sure "OperationalException" is the right one here?
That exception is assumed to stop the bot - so it completely restarts - though i'd not do that just because there's no ticks ... which could very well happen on random, low-volume pairs (?)

Reply:
Do you have any recommendation what to use instead?